src/EnergyPlus/UnitarySystem.cc

```diff
@@ -1222,6 +1205,25 @@ namespace UnitarySystems {
 
         // PURPOSE OF THIS SUBROUTINE:
         // This subroutine checks for proper set point at control node.
+        constexpr static std::array<std::string_view, 3> coilTypes = {"cooling", "heating", "supplemental"};
+        bool SetPointErrorFlag = false;
+
+        if (ControlNode == 0) {
+            if (this->m_ControlType == UnitarySysCtrlType::Setpoint) {
+                int coilOutNode = this->CoolCoilOutletNodeNum;
+                if (CoilType == HeatingCoil) coilOutNode = this->HeatCoilOutletNodeNum;
+                if (CoilType == SuppHeatCoil) coilOutNode = this->SuppCoilOutletNodeNum;
```

Comment:
I suspect this also would be faster but didn't think of it in time.

    int coilOutNode = (CoilType == CoolingCoil) ? this->CoolCoilOutletNodeNum : ((CoilType == HeatingCoil) ? this->HeatCoilOutletNodeNum
                    : this->SuppCoilOutletNodeNum);